### ISSUE TITLE
image tag does not support svg

### ISSUE THREAD

**Issue by trilogysci** [SVGKit/SVGKit#159]
The image tag seems to support png files but not svg files.

as in the following

<?xml version="1.1" encoding="UTF-8"?>
<svg width="480px" height="480px" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1">
    <title>Note</title>
    <g>
        <path d="M240,476 C370,476 476,370 476,240 C476,109 370,4 240,4 C109,4 4,109 4,240 C4,370 109,476 240,476" fill="rgb(230,250,210)"></path>
        <path d="M179,311 C179,311 176,305 157,305 C138,305 108,314 108,349 C108,370 132,383 154,383 C176,383 202,364 202,342 C202,318 202,201 202,201 C202,201 268,177 330,177 L330,277 C330,277 318,273 307,273 C288,273 255,287 259,317 C263,347 291,351 306,351 C330,351 354,331 354,311 C354,295 354,96 354,96 C354,96 232,99 179,129 Z M179,311" fill="darkgreen"></path>
    </g>
    <image xlink:href="https://upload.wikimedia.org/wikipedia/commons/9/9a/PNG_transparency_demonstration_2.png" x="80" y="0" width="200px" height="200px" />
    <image xlink:href="https://upload.wikimedia.org/wikipedia/commons/0/02/SVG_logo.svg" width="200px" height="200px" x="0" y="80"/>
</svg>

**Comment by trilogysci** [comment]
The following modification to SVGImageElement.m seems to do the trick.
It also supports custom url scheme bundle: to designate files in the bundle.

``` objectivec
-(NSString *)findFileInBundleByName: (NSString*) name
{
    NSParameterAssert(name != nil);
    /** Apple's File APIs are very very bad and require you to strip the extension HALF the time.

     The other HALF the time, they fail unless you KEEP the extension.

     It's a mess!
     */
    NSString *newName = [name stringByDeletingPathExtension];
    NSString *extension = [name pathExtension];
    if ([@"" isEqualToString:extension]) {
        extension = @"svg";
    }

    /** First, try to find it in the project BUNDLE (this was HARD CODED at compile time; can never be changed!) */
    NSString *pathToFileInBundle = nil;
    NSBundle *bundle = [NSBundle mainBundle];
    if( bundle != nil )
    {
        pathToFileInBundle = [bundle pathForResource:newName ofType:extension];
    }

    /** Second, try to find it in the Documents folder (this is where Apple expects you to store custom files at runtime) */
    NSString* pathToFileInDocumentsFolder = nil;
    NSString* pathToDocumentsFolder = [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject];
    if( pathToDocumentsFolder != nil )
    {
        pathToFileInDocumentsFolder = [[pathToDocumentsFolder stringByAppendingPathComponent:newName] stringByAppendingPathExtension:extension];
        if( [[NSFileManager defaultManager] fileExistsAtPath:pathToFileInDocumentsFolder])
        ;
        else
        pathToFileInDocumentsFolder = nil; // couldn't find a file there
    }

    if( pathToFileInBundle == nil
       && pathToFileInDocumentsFolder == nil )
    {
        DDLogCWarn(@"[%@] MISSING FILE (not found in App-bundle, not found in Documents folder), COULD NOT CREATE DOCUMENT: filename = %@, extension = %@", [self class], newName, extension);
        return nil;
    }

    /** Prefer the Documents-folder version over the Bundle version (allows you to have a default, and override at runtime) */
    return pathToFileInDocumentsFolder == nil ? pathToFileInBundle : pathToFileInDocumentsFolder;
}

- (CALayer *) newLayer
{
    CALayer* newLayer = [[CALayer alloc] init];

    [SVGHelperUtilities configureCALayer:newLayer usingElement:self];

    /** transform our LOCAL path into ABSOLUTE space */
    CGRect frame = CGRectMake(_x, _y, _width, _height);
    frame = CGRectApplyAffineTransform(frame, [SVGHelperUtilities transformAbsoluteIncludingViewportForTransformableOrViewportEstablishingElement:self]);
    newLayer.frame = frame;
    NSData *imageData = nil;
    NSString *path = nil;
    if([_href hasPrefix:@"bundle:"]) {
        path = [self findFileInBundleByName: [_href substringFromIndex:[@"bundle:" length]]];
        imageData = [NSData dataWithContentsOfFile: path];
    } else {
        imageData = [NSData dataWithContentsOfURL:[NSURL URLWithString:_href]];
    }
    SVGImageRef image = [SVGImage imageWithData:imageData];
    if(image ) {
        newLayer.contents = (id)SVGImageCGImage(image);
    } else if(imageData){
        SVGKImage *simage = nil;
        SVGKSource* source = [SVGKSource sourceFromData: imageData];
        if(source) {
            simage = [SVGKImage imageWithSource: source];
            if(simage) {
                image = simage.UIImage;
                if(image) {
                    newLayer.contents = (id)SVGImageCGImage(image);
                }
            }
        }
    }
```

**Comment by adamgit** [comment]
If this works, submit a pull request woth just this change, and we can
merge it.

**Comment by adamgit** [comment]
@trilogysci  Pull request?

Also - is that example SVG image public domain? Can you add it to the demo project?

**Comment by adamgit** [comment]
This caused a lot of problems. I've rewritten it to work more smoothly and integrate with the recent changes to local/relative file URLs.

Also, I made a new demo SVG, since I wasn't sure if we could use the one above.

Fixed by 90cc88b97a427a3450cc600f19f44ece7bab0bf0
